### PR TITLE
push_down_ne

### PR DESCRIPTION
**Pull request by guojidan**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [x] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #559 .

# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: common/models/src/predicate/domain.rs [hunk 1]
```diff
@@ -692,6 +703,13 @@ impl Range {
     fn overlaps(&self, other: &Self) -> Result<bool> {
         self.check_type_compatibility(other)?;
         if self.low <= other.high && other.low <= self.high {
+            // like (-∞, 2) and (2, +∞) is not overlap
+            if self.high == other.low
+                && self.high.bound == Bound::Below
+                && other.low.bound == Bound::Above
+            {
+                return Ok(false);
+            }
```
Comment (yukkit) [inline common/models/src/predicate/domain.rs:712]: nice find!

**Review by yukkit — APPROVED** [review]

**Comment by yukkit** [comment]
It works for me, but it would be better if you can add related test cases

**Comment by guojidan** [comment]
> It works for me, but it would be better if you can add related test cases

ok, I add some test case for it.

File: common/models/src/predicate/domain.rs (hunk 1 above)
Comment (guojidan) [reply common/models/src/predicate/domain.rs:712]: : )

**Comment by guojidan** [comment]
> > It works for me, but it would be better if you can add related test cases
> 
> ok, I add some test case for it.

I think adding test cases for the Range:: ne function is sufficient

**Review by yukkit — APPROVED** [review]

**Pull request merged**